Source/WebSocket.swift

```diff
@@ -1668,13 +1668,17 @@ public class WebSocket: NSObject {
     public init(request: NSURLRequest, subProtocols : [String] = []){
         opened = true
         ws = InnerWebSocket(request: request, subProtocols: subProtocols, stub: false)
+        super.init()
+        ws.eclose = { [unowned self] in
```

Comment:
Yeah, that's a good idea. I wondered earlier why `init()` is even allowed (considering it takes no parameters and wouldn't open a websocket)? But on second thoughts, there could be use cases for it, when used with one of the `open` variants.

I'm not sure whether `init()` needs to be a designated initializer, so I'd probably go with it being a convenience initializer (to avoid duplicate code):
```swift
    /// Create a WebSocket connection from an NSURLRequest; Also include a list of protocols.
    public init(request: NSURLRequest, subProtocols : [String] = []){
        let hasURL = request.URL != nil
        opened = hasURL
        ws = InnerWebSocket(request: request, subProtocols: subProtocols, stub: !hasURL)
        super.init()
        ws.eclose = { [unowned self] in
            self.opened = false
        }
    }
    /// Create a WebSocket object with a deferred connection; the connection is not opened until the .open() method is called.
    public convenience override init(){
        self.init(request: NSURLRequest(), subProtocols: [])
    }
```